client-interface/src/main/java/com/flipkart/grayskull/GrayskullClient.java

```diff
@@ -0,0 +1,15 @@
+package com.flipkart.grayskull;
+
+import com.flipkart.grayskull.hooks.RefreshHookHandle;
+import com.flipkart.grayskull.hooks.SecretRefreshHook;
+import com.flipkart.grayskull.models.SecretValue;
+
+public interface GrayskullClient extends AutoCloseable {
+
+    SecretValue getSecret(String secretRef);
+
+    RefreshHookHandle registerRefreshHook(String secretRef, SecretRefreshHook hook);
+
+    @Override
+    void close();
```

Comment:
why is this part of interface?

AFAIK, you don't have to specify overrides here. only in impls .